pkg/controller/gcp/database/operations.go

```diff
@@ -141,7 +141,12 @@ func (h *localHandler) updateConnectionSecret(ctx context.Context) (*corev1.Secr
 		if _, found := s.Data[runtimev1alpha1.ResourceCredentialsSecretPasswordKey]; !found {
 			s.Data[runtimev1alpha1.ResourceCredentialsSecretPasswordKey] = []byte(password)
 		}
-		s.Data[runtimev1alpha1.ResourceCredentialsSecretEndpointKey] = secret.Data[runtimev1alpha1.ResourceCredentialsSecretEndpointKey]
+		if _, found := secret.Data[v1alpha2.PrivateIPKey]; found {
+			s.Data[v1alpha2.PrivateIPKey] = secret.Data[v1alpha2.PrivateIPKey]
+		}
+		if _, found := secret.Data[v1alpha2.PublicIPKey]; found {
+			s.Data[v1alpha2.PublicIPKey] = secret.Data[v1alpha2.PublicIPKey]
+		}
```

Comment:
I don't think there's much need for these conditionals. If `s.Data[v1alpha2.PrivateIPKey]` was not found in the map a lookup against that key would return the zero value (a nil byte slice), thus assigning `secret.Data[v1alpha2.PrivateIPKey]` to a nil byte slice (its zero value).